e2e/scaffolding/globalHooks.ts

```diff
@@ -0,0 +1,49 @@
+import { cryptoWaitReady } from "@polkadot/util-crypto";
+import { ExtrinsicHelper } from "./extrinsicHelpers";
+import { fundingSources, getFundingSource, getRootFundingSource, getSudo } from "./funding";
+import { TEST_EPOCH_LENGTH, getNonce, setEpochLength } from "./helpers";
+import { isDev } from "./env";
+
+const SOURCE_AMOUNT = 100_000_000_000_000n;
+
+async function fundAllSources() {
+  const root = getRootFundingSource().keys;
+  console.log("Root funding source: ", root.address);
+  const nonce = await getNonce(root);
+  await Promise.all(fundingSources.map((dest, i) => {
+    return ExtrinsicHelper.transferFunds(root, getFundingSource(dest), SOURCE_AMOUNT).signAndSend(nonce + i);
+  }));
+}
+
+async function devSudoActions() {
+  // Because there is only one sudo, these actions must take place globally
+  const sudo = getSudo().keys;
+  await setEpochLength(sudo, TEST_EPOCH_LENGTH);
+}
+
+async function drainAllSources() {
+  const root = getRootFundingSource().keys;
+  await Promise.all(fundingSources.map((source, i) => {
+    return ExtrinsicHelper.emptyAccount(getFundingSource(source), root).signAndSend();
+  }));
+}
+
+export async function mochaGlobalSetup() {
+  console.log('Global Setup Start');
+  await cryptoWaitReady();
+  await ExtrinsicHelper.initialize();
+  await fundAllSources();
+
+  // Sudo is only when not on Testnet
+  if (isDev()) await devSudoActions();
+
+  console.log('Global Setup Complete');
+}
+
+export async function mochaGlobalTeardown() {
+  console.log('Global Teardown Start');
+  await drainAllSources();
```

Comment:
Update to above:  I did actually do another test run by `rm -rf /tmp/frequency && make start` and then `make e2e-test` and that's when I saw the test fail the deep-equal + the leftover accounts with 5 UNIT.

Reply:
It will only drain the `fundingSources` from `funding.tx`. The additional accounts generated by the test itself may not be reaped. (Although it might be a good idea to set something else up to do that).

Can you check if it was one of the `fundingSources` remaining?

Reply:
Added some code to drain as many of the accounts as possible.